### DISCUSSION TITLE
Search bar for nodes

### DISCUSSION THREAD

**Discussion by Tokarzewski** [IndiePython/nodezator#27] [Feature Request]
Search bar for nodes that show up after selecting spacebar.

**Comment by KennedyRichard** [comment]
Good one, just like the one on Blender3D, right? But instead of a general search for utilities it would search for nodes. Just added your request to the list of requested features #13 . Since the services needed to implement this feature are already available in Nodezator it should be pretty quick to implement too. I'll probably include it on the next round of changes.

**Comment by djbnet** [comment]
Would love to see this implemented! Microsoft Copilot is decent at helping beginners navigate Nodezator, but it doesn't know the hierarchy well enough, hindering workflow. Coming from Unreal Blueprints, it would help a lot!

**Reply by KennedyRichard** [reply]
Thank you for your feedback djbnet, it is much appreciated. I mentioned the need and possibility of implementing this feature a few times in recent discussions.

It wouldn't even require much work actually, because most of the UI tools and underlying services needed for this feature are already available (that is, we do have front end and back end tools to implement it).

Nodezator development is paused, but only temporarily, due to my work in other projects. I do have a schedule for getting back into working on it, although I didn't publish yet: I expect to get back to working on Nodezator in a few months, hopefully launching a new version by the end of the year or beginning of the next one. I intend to implement the search bar then.

Sorry for the long deadlines, it is just a consequence of being a solo developer with 03 projects (Nodezator and another two). Don't worry though, as I'm not the kind to start new projects haphazardly without caring for my current ones, I'm really sticking to the 03 projects, working on them in a sort of rotation. Also, this never prevented Nodezator for getting proper development and maintenance, new versions and patches at least a couple times each year.
